t/t0029-filemap-cmd.t

```diff
@@ -0,0 +1,244 @@
+'
+test_expect_success HAVE_JQ 'show fileref' '
+	flux filemap list --fileref | jq .
+'
+test_expect_success 'test file can be read through content cache' '
+	flux filemap get -vv -C copydir
+'
+test_expect_success 'copy is a symlink with expected target' '
+	readlink testfile4 >link.exp &&
+	readlink copydir/testfile4 >link.out &&
+	test_cmp link.exp link.out
+'
+test_expect_success 'unmap test file' '
+	flux filemap unmap
+'
+test_expect_success 'map file by absolute path' '
+	flux filemap map /etc/group
+'
+test_expect_success 'test file lists with relative path' '
+	cat >list.exp <<-EOT &&
+	etc/group
+	EOT
+	flux filemap list >list.out &&
+	test_cmp list.exp list.out
+'
+test_expect_success 'unmap test file' '
+	flux filemap unmap
+'
+test_expect_success 'create test directory' '
+	mkdir testfile5
+'
+test_expect_success 'map test file' '
+	flux filemap map ./testfile5
+'
+test_expect_success 'test file can be read through content cache' '
+	flux filemap get -vv -C copydir
+'
+test_expect_success 'copy is a directory' '
+	test -d copydir/testfile5
+'
+test_expect_success 'unmap test file' '
+	flux filemap unmap
+'
+test_expect_success 'map testfile with tags=red,blue' '
+	flux filemap map --tags=red,blue ./testfile
+'
+test_expect_success 'map testfile with tags=green' '
+	flux filemap map --tags=green ./testfile
+'
+test_expect_success 'map testfile with tags=green (again)' '
+	flux filemap map --tags=green ./testfile
+'
+test_expect_success 'map testfile2 with tags=green' '
+	flux filemap map --tags=green ./testfile2
+'
+test_expect_success 'list tags=red,blue,green reports two files' '
+	test $(flux filemap list --tags=red,blue,green | wc -l) -eq 2
+'
+test_expect_success 'unmap red tag' '
+	flux filemap unmap --tags=red
+'
+test_expect_success 'list tags=red,blue,green reports two files' '
+	test $(flux filemap list --tags=red,blue,green | wc -l) -eq 2
+'
+test_expect_success 'unmap tags=blue,green' '
+	flux filemap unmap --tags=blue,green
+'
+test_expect_success 'flux filemap list reports no files' '
+	test $(flux filemap list --tags red,blue,green | wc -l) -eq 0
+'
+test_expect_success 'map test file' '
+	rm -f copydir/testfile &&
+	flux filemap map ./testfile
+'
+test_expect_success 'modify mapped test file without reducing its size' '
+	dd if=/dev/zero of=testfile bs=4096 count=1 conv=notrunc
+'
+test_expect_success 'content change should cause an error' '
+	rm -f copydir/testfile &&
+	test_must_fail flux filemap get -C copydir 2>changed.err &&
+	grep changed changed.err
+'
+test_expect_success 'drop cache and unmap test file' '
+	flux content dropcache &&
+	flux filemap unmap
+'
+test_expect_success 'map test file' '
+	rm -f copydir/testfile &&
+	flux filemap map ./testfile
+'
+test_expect_success 'truncate mapped test file' '
+	cp /dev/null testfile
+'
+test_expect_success 'size reduction should cause an error' '
+	rm -f copydir/testfile &&
+	test_must_fail flux filemap get -C copydir 2>reduced.err &&
+	grep changed reduced.err
+'
```

Comment:
just for coverage, increase file and make sure still works?

Reply:
also possibly relevant given my prior perms question, test coverage of change in permissions?

Reply:
That turns out to be hard to test because we cannot force a re-stat(2) of the file until the rate limit runs time (5s) runs out.  Anyway, recovery after an error has been detected is not really part of the design.  In fact we might want to unmap the file in that case and invalidate the other cache entries at some point.